notebooks/generative-ai/lib/elasticsearch_chat_message_history.py

```diff
@@ -0,0 +1,81 @@
+import json
+import logging
+from typing import List
+from elasticsearch import ApiError, Elasticsearch
+
+from langchain.schema import BaseChatMessageHistory
+from langchain.schema.messages import BaseMessage, _message_to_dict, messages_from_dict
+
+logger = logging.getLogger(__name__)
+
+class ElasticsearchChatMessageHistory(BaseChatMessageHistory):
+    """Chat message history that stores history in Elasticsearch.
+
+    Args:
+        client: Elasticsearch client.
+        index: name of the index to use.
+        session_id: arbitrary key that is used to store the messages
+            of a single chat session.
+    """
+
+    def __init__(
+        self,
+        client: Elasticsearch,
+        index: str,
+        session_id: str,
+    ):
+        self.client: Elasticsearch = client
+        self.index: str = index
+        self.session_id: str = session_id
+
+        if not client.indices.exists(index=index):
```

Comment:
add a debug message to say that the indices are being created / skipped.

Reply:
Done.